dashboard/app/models/unit_group.rb

```diff
@@ -261,6 +261,14 @@ def update_scripts(new_units, alternate_units = nil)
     end
 
     units_to_remove.each do |unit|
+      # Units that are not in a unit group need to have these fields set in order to determine course type and visibility of course
+      unit.update!(
+        published_state: (unit.published_state ? unit.published_state : published_state),
```

Comment:
super nit: you could just do `unit.published_state || published_state` here. Feel free to ignore if you don't think it's cleaner

Reply:
Thats a good call out. I'm gonna go ahead with it as is for now tho.